### DISCUSSION TITLE
Password hashing / encryption

### DISCUSSION THREAD

**Discussion by gtamas** [strapi/strapi#6187] [Legacy Topics]
I'm just curious, how does Strapi store user passwords?
I mean, I can see they are encrypted, but is there documentation on how this encryption / hashing happens? What algorithms are used etc (I can see that strapi-plugin-users-permissions is based on bcrypt).

**Comment by lauriejim** [comment]
Hello!

Here is the code of the register function that encrypt the password - [code](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/controllers/Auth.js#L461)

And here is the function that is called - [code](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/services/User.js#L77)

**Reply by gtamas** [reply]
Ok Thank you!
